cmd/nerdctl/network_inspect.go

```diff
@@ -58,50 +61,75 @@ func networkInspectAction(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-
-	netMap, err := e.NetworkMap()
+	mode, err := cmd.Flags().GetString("mode")
+	if err != nil {
+		return err
+	}
+	if mode != "native" && mode != "dockercompat" {
+		return fmt.Errorf("unknown mode %q", mode)
+	}
+	format, err := cmd.Flags().GetString("format")
 	if err != nil {
 		return err
 	}
 
-	result := make([]interface{}, len(args))
-	for i, name := range args {
+	var result []interface{}
+	var errs []error
+
+	walker := netwalker.NetworkWalker{
+		Client: e,
+		OnFound: func(ctx context.Context, found netwalker.Found) error {
+			if found.MatchCount > 1 {
+				return fmt.Errorf("multiple IDs found with provided prefix: %s", found.Req)
+			}
+			r := &native.Network{
+				CNI:           json.RawMessage(found.Network.Bytes),
+				NerdctlID:     found.Network.NerdctlID,
+				NerdctlLabels: found.Network.NerdctlLabels,
+				File:          found.Network.File,
+			}
+			switch mode {
+			case "native":
+				result = append(result, r)
+			case "dockercompat":
+				compat, err := dockercompat.NetworkFromNative(r)
+				if err != nil {
+					return err
+				}
+				result = append(result, compat)
+			}
+			return nil
+		},
+	}
+
+	for _, name := range args {
 		if name == "host" || name == "none" {
-			return fmt.Errorf("pseudo network %q cannot be inspected", name)
+			errs = append(errs, fmt.Errorf("pseudo network %q cannot be inspected", name))
+			continue
 		}
-		l, ok := netMap[name]
-		if !ok {
-			return fmt.Errorf("no such network: %s", name)
+		n, err := walker.Walk(cmd.Context(), name)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		} else if n == 0 {
+			errs = append(errs, fmt.Errorf("no such network: %s", name))
 		}
+	}
 
-		r := &native.Network{
-			CNI:           json.RawMessage(l.Bytes),
-			NerdctlID:     l.NerdctlID,
-			NerdctlLabels: l.NerdctlLabels,
-			File:          l.File,
-		}
-		mode, err := cmd.Flags().GetString("mode")
+	if len(result) > 0 {
+		err = formatter.FormatSlice(format, cmd.OutOrStdout(), result)
 		if err != nil {
-			return err
-		}
-		switch mode {
-		case "native":
-			result[i] = r
-		case "dockercompat":
-			compat, err := dockercompat.NetworkFromNative(r)
-			if err != nil {
-				return err
-			}
-			result[i] = compat
-		default:
-			return fmt.Errorf("unknown mode %q", mode)
+			errs = append(errs, err)
 		}
 	}
-	format, err := cmd.Flags().GetString("format")
-	if err != nil {
-		return err
+	for _, err := range errs {
+		logrus.Error(err)
+	}
```

Comment:
nit: this can be moved into the `if len(errs) > 0` since they're all code only related to errors.

Reply:
done